### PR TITLE
ANNULUS-307: Wallet: Update the next button in Stepper Screen to support form-checking

### PR DESCRIPTION
**Pull request by lukorito**
## Description

Wallet: Update the next button in Stepper Screen to support form-checking

Fixes # (issue)

## Type of change

Please delete options that are not relevant.
-   [x] New feature (non-breaking change which adds functionality)
## How Has This Been Tested?

Manually

## Checklist:

-   [ ] My code follows the style guidelines of this project
-   [ ] I have performed a self-review of my own code
-   [ ] I have commented my code, particularly in hard-to-understand areas
-   [ ] I have made corresponding changes to the documentation
-   [ ] My changes generate no new warnings
-   [ ] I have added tests that prove my fix is effective or that my feature works
-   [ ] New and existing unit tests pass locally with my changes
-   [ ] Any dependent changes have been merged and published in downstream modules
-   [ ] I have checked my code and corrected any misspellings
-   [ ] I have added screenshots

### CODE REVIEW

File: lib/v2/shared/providers/stepper_navigation_control_prover.dart [hunk 1]
```diff
@@ -0,0 +1,36 @@
+// Package imports:
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+
+enum StepperNavigationControl {
```
Comment (KalervoHyyppa) [inline lib/v2/shared/providers/stepper_navigation_control_prover.dart:4]: instead of an enum, make a freezed class similar to `StepperScreenState`

Comment (KalervoHyyppa) [inline lib/v2/shared/providers/stepper_navigation_control_prover.dart:4]: instead of an enum, make a freezed class similar to `StepperScreenState`

**Review by KalervoHyyppa — COMMENTED** [review]
Add checks to the recovery seed phrase input as well

File: lib/v2/onboarding/widgets/pages/create_pin_page.dart [hunk 2]
```diff
@@ -26,18 +29,27 @@ class CreatePinPage extends HookWidget {
   }) : super(key: key);
 
   @override
-  Widget build(BuildContext context) {
+  Widget build(BuildContext context, WidgetRef ref) {
     final focusNode = useFocusNode();
     final isPinValid = useState(false);
+    final nextButtonNotifier = ref.watch(stepperNavigationControlProvider.notifier);
+    useEffect(() {
```
Comment (KalervoHyyppa) [inline lib/v2/onboarding/widgets/pages/create_pin_page.dart:36]: please add a

```
return (){
    null
}
```

to fix build errors

**Review by KalervoHyyppa — CHANGES_REQUESTED** [review]
If I type in a correct pin but then type and incorrect pin, I can still go forward. Also can go forward on the confirm pin page wihtout type pin as well. See video

https://github.com/Topl/ribn/assets/51681634/f03a58f4-e7e4-47db-99ea-9ed5de0a84d1

**Comment by lukorito** [comment]
> If I type in a correct pin but then type and incorrect pin, I can still go forward. Also can go forward on the confirm pin page wihtout type pin as well. See video
> 
>  screen-20230911-155144.mp4

Fixed this as well

**Review by KalervoHyyppa — APPROVED** [review]
lgtm

**Pull request merged**